### PR TITLE
The CodeQL analysis is broken due to the large content of the SARIF file

### LINKED ISSUE #10203: The CodeQL analysis is broken due to the large content of the generated SARIF file
### Describe the bug

The CodeQL analysis is broken due to the large content of the generated SARIF file.

### Version

17.0.0

### Anything else?

Ref: This is a follow up of https://github.com/keycloak/keycloak/pull/10108#issuecomment-1036180254

### PR DESCRIPTION
**Pull request by abstractj**
The CodeQL analysis is broken due to the large content of the generated
SARIF file.

The issue was originally caused by high number of flows paths per alert
generated by the LDAP federation module. That was identified taking the
SARIF file generated and running:

```
jq '.runs[0].results | map({query_id: .rule.id, numPaths: .codeFlows |
length})' java.sarif

```

Together, we managed to reduce the number of flows paths, and also worked
on the optimizations to skip some paths.

Co-authored-by: Bruno Oliveira da Silva <bruno@abstractj.com>

Closes #10203

### How to test it

The testing must be done manually. For security reasons, GitHub won't give permissions to a pull-request to read `GITHUB_TOKEN` and write security events. So, it is expected that all the checks in this PR for CodeQL will fail. 

1. Fetch the files submitted for this PR and push to the main branch of your fork
2. Submit a couple of PRs to your fork to see the checks passing

### Working PoC
- https://github.com/keycloak-poc/keycloak/pull/9

### CODE REVIEW

**Comment by abstractj** [comment]
@JoshuaMulliken FYI

**Comment by abstractj** [comment]
Checks are failing because we have restricted access to third-party apps. I will revisit the current approach and update this PR.

**Comment by abstractj** [comment]
Closing it until it gets fixed, so we don't waste people's time.

**Comment by andreaTP** [comment]
Thanks for the update @abstractj 

It is my understanding that we "should" be able to re-use the `github/codeql-action` with proper parameters instead of rolling our own scripts, would you like to detail what is missing?

We should make sure that we use `wait-for-processing` to fail and not ignore extra results, but we also need this fix to be in before it can be used:
https://github.com/github/codeql-action/issues/963

**Comment by abstractj** [comment]
@andreaTP unfortunately, the `github/codeql-action` has a limited configuration for our use case, mainly with monorepos – there's a great discussion about this on https://github.com/github/codeql-action/issues/820. Using CodeQL cli gives us more flexibility to remove unnecessary flow paths, and instruct the scanner to skip folders that we don't need.

@JoshuaMulliken and I have been investigating a proper fix for Keycloak for almost 2 weeks, but if you have a working solution with `github/codeql-action` to share with us, I don't see any problem about experimenting it.

**Comment by abstractj** [comment]
@andreaTP awesome, please let me know if you would like to share a working solution.

**Comment by andreaTP** [comment]
@abstractj I'm sorry but I don't have enough context to look into this issue at the moment, nor I have a working solution.

My understanding was that you and @JoshuaMulliken were on it.
Happy to participate if you believe that my contribution can be of any value.

**Comment by JoshuaMulliken** [comment]
@abstractj @abstractj since we have a working solution right now, I recommend that we roll with it. Once we are in a better position we can adjust & utilize another system

**Comment by abstractj** [comment]
@andreaTP no problem Andrea, let's wait for more feedback on this.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -7,17 +7,19 @@ name: "CodeQL"
 
 on:
 # Disable for push and pull_request until https://github.com/keycloak/keycloak/issues/10203 is resolved
-#  push:
-#    branches: [main]
-#  pull_request:
+  push:
+    branches: [main]
+  pull_request:
```
Comment (andreaTP) [inline .github/workflows/codeql-analysis.yml:12]: I might be missing something, but given:

> For security reasons, GitHub won't give permissions to a pull-request to read `GITHUB_TOKEN` and write security events. So, it is expected that all the checks in this PR for CodeQL will fail.

after this PR is merged to `main` are we going to have it working on PRs or should we only do the scan on `main`?

Comment (abstractj) [reply .github/workflows/codeql-analysis.yml:12]: @andreaTP yes, please check the working PoC added to the description. Also, I suggest experimenting it by copying those files to your fork.

**Review by stianst — APPROVED** [review]

**Pull request merged**